src/main.cc

```diff
@@ -2,8 +2,24 @@
 
 // The purpose of this file is to check that all headers compile
 
+#if CLAP_VERSION_LT(CLAP_VERSION_MAJOR, CLAP_VERSION_MINOR, CLAP_VERSION_REVISION)
+SOFTWARE ERROR: CLAP_VERSION_LT is inconsistent
```

Comment:
You can use the `#error` pragma for this. That prints the erorr message instead of resulting in a probably less readable parsing error.

Reply:
I thought some compilers didn’t support the error pragma - especially older msvc - hence the choice. But happy to change it sure!

Reply:
Oh but in this spot it doesn't matter anyway since its a c++ file we just compile as a test. (I originally had these in the header then moved them). Anyway pushed change.

Reply:
Some older compilers indeed may not support it, but then you'd just get an unsupported pragma warning or error with a reference to the line.  That's probably still better than the compiler just chugging along and trying to make sense of the file with those error messages as plain text. Microsoft's documentation doesn't go back before Visual Studio 2015, but that version does have `#error`. And if one compiler doesn't support it, it's probably MSVC.

Reply:
#error is supported back until Visual Studio 98, at least ... I'd have to fire up a virtual XP to check whether it works in Visual C++ 1.52 as well, but am pretty sureit does. I seem to remember that it already worked in Microsoft C 1.00 which I had in 1983 (I got no possibility to verify that anymore, though). The whole #pragma stuff is something that came in later (and where Microsoft and GNU C/C++ have severe syntactical disagreements).

Reply:
Yeah but also in this context - where I added the error to the test case not the header file - our compiler set is smaller so they *definitely* support it.

Anyway curious how folks feel about the version as submitted here now. I think it meets all review constraints yeah?